task.go

```diff
@@ -325,7 +325,12 @@ func (t *task) Delete(ctx context.Context, opts ...ProcessDeleteOpts) (*ExitStat
 		return nil, fmt.Errorf("task must be stopped before deletion: %s: %w", status.Status, errdefs.ErrFailedPrecondition)
 	}
 	if t.io != nil {
-		t.io.Close()
+		// io.Wait locks for restored tasks on Windows unless we call
+		// io.Close first (https://github.com/containerd/containerd/issues/5621)
+		// in other cases, preserve the contract and let IO finish before closing
+		if t.client.runtime == fmt.Sprintf("%s.%s", plugin.RuntimePlugin, "windows") {
```

Comment:
In order to not reintroduce https://github.com/containerd/containerd/issues/5621 which was fixed by https://github.com/containerd/containerd/pull/8643, we leave the `io.Close()` call here, on Windows only. (for now, until the underlying issue re: `io.Wait()` locking for restored tasks on Windows)

Reply:
Looks good to me. The `Delete` should be called after the task has been stopped. 
The kernel or shim will ensure all the processes that are created by this task are killed. Just needs to ensure that the `t.io.Cancel()` won't close the fifo file (REF: https://github.com/containerd/containerd/pull/8334).